datafusion/core/src/datasource/physical_plan/parquet/statistics.rs

```diff
@@ -211,32 +394,25 @@ pub(crate) fn min_statistics<'a, I: Iterator<Item = Option<&'a ParquetStatistics
     data_type: &DataType,
     iterator: I,
 ) -> Result<ArrayRef> {
-    let scalars = iterator
-        .map(|x| x.and_then(|s| get_statistic!(s, min, min_bytes, Some(data_type))));
-    collect_scalars(data_type, scalars)
+    match get_statistics_iter!(iterator, min, min_bytes, Some(data_type)) {
```

Comment:
This is pretty neat @xinlifoobar 

As this is currently implemented, it has quite a bit of repetition (e.g. the clauses for                             `ParquetStatistics::ByteArray(s)` in the macro above

What do you think of a slightly different mode where the iteration over the statistics values and the array creation looked different

Something like
```rust
match data_type {
  DataType::Int8 => {
    // get an iterator over int32 stored in 
    let int32_iter = get_statistics_iter!(iterator, min, ParquetStatistics::Int32)
    // convert int32 to int8, ignoring 
    let int8_iter = int32_iter.map(|v| if let Ok(v) = v.try_into:<i8>() { Some(v) } else { None });
    Int8Array::from_iter(int8_iter)
  } 
  // similarly for other types
 ....
  DataType::Utf8 => {
    // get an iterator over bytes stored
    let byte_iter = get_statistics_iter!(iterator, min_bytes, ParquetStatistics::ByteArray)
    // convert bytes to utf8 strings
    let string_array = byte_iter.map(|v| if let Ok(v) = str::from_utf8....);
    StringArray::from_iter(int8_iter)
  } 
 ...
```

Another reason I suggest this structure is that I think it would make it easier to support extracting statistics from multiple files in the future more easily

Reply:
This looks good to me but without macros the repetations, between `min` are `max`  still... Do you suggest to have another macro to wrap all above and reuse it in min statistic and max statistic?

Reply:
I played around with this code this afternoon and here is what I came up with: https://github.com/apache/datafusion/pull/10715

What do you think?